File: samples/dart/04primality/prime.dart
import "dart:io";
import "dart:math";
import "package:sprintf/sprintf.dart";


bool is_prime(int number) {
    int limit = sqrt(number).toInt();
    for (int count = 2; count <= limit; count++) {
        if (number % count == 0) {
            return false;
        }
    }
    return true;
}


void main(List<String> arguments) {
    if (arguments.length < 1) {
        stderr.write("Error: Usage dart prime.dart <number> [<number> ...]\n");
        exit(1);
    }

    for (String arg in arguments) {
        int number = 0;
        try {
            number = int.parse(arg);
        }
        on FormatException {
            stderr.write(sprintf("Error: Expected to receive an integer. Received '%s'\n", arg));
            exit(2);
        }
        print(sprintf("%d => %s", [number, is_prime(number) ? "true" : "false"]));
    }
}

File: samples/lua/04primality/prime.lua
function printf(s, ...)
    return io.write(s:format(...))
end

function eprintf(s, ...)
    return io.stderr:write(s:format(...))
end

function is_prime (number)
    local limit = math.sqrt(number)
    for n = 2,limit do
        if number % n == 0 then
            return false
        end
    end
    return true
end

if #arg == 0 then
    eprintf("Error: Usage: lua prime.lua <number> [<number> ...]\n")
    os.exit(1)
end

for i=1,#arg do
    number = tonumber(arg[i])
    if number == nil then
        eprintf("Error: Expected to receive an integer. Received '%s'\n", arg[i])
        os.exit(2)
    end
    printf("%s => %s\n", arg[i], is_prime(number) and 'true' or 'false')
end
